[Bug Report] SB: SQL on backend with comma table as JOIN

<!--
    Note: In order to better solve your problem, please refer to the template to provide complete information, accurately describe the problem, and the incomplete information issue will be closed.
-->
## Bug report

When join on view definition is `, C_Currency cto`

![Image](https://github.com/user-attachments/assets/888edae3-0bad-47d1-aa69-0d464b96f15d)


Generates error when executing sql due to adding more joins for display columns.

#### Screenshot or Gif

![Image](https://github.com/user-attachments/assets/a2a8b3e5-b036-423a-abd3-69b50f2931e5)

#### Expected behavior
A clear and concise description of what you expected to happen.

#### Additional context
The backend log
```
===========> UserInterface.listBrowserItems: org.postgresql.util.PSQLException: ERROR: invalid reference to FROM-clause entry for table "wh"
  Hint: There is an entry for table "wh", but it cannot be referenced from this part of the query.
  Position: 217, SQL=SELECT COUNT(*)  FROM WH_Withholding wh LEFT JOIN WH_Definition whd ON (whd.WH_Definition_ID=wh.WH_Definition_ID) , C_Currency cto LEFT JOIN AD_Ref_List AS WH_DocStatus_AD_Ref_List ON(WH_DocStatus_AD_Ref_List.Value = wh.DocStatus AND WH_DocStatus_AD_Ref_List.AD_Reference_ID = 131) LEFT JOIN AD_Ref_List_Trl AS WH_DocStatus_AD_Ref_List_Trl ON(WH_DocStatus_AD_Ref_List_Trl.AD_Ref_List_ID = WH_DocStatus_AD_Ref_List.AD_Ref_List_ID AND WH_DocStatus_AD_Ref_List_Trl.AD_Language = 'es_MX') LEFT JOIN C_Invoice AS WH_SourceInvoice_ID_C_Invoice ON(WH_SourceInvoice_ID_C_Invoice.C_Invoice_ID = wh.SourceInvoice_ID) WHERE wh.AD_Client_ID IN(0,1000001) AND wh.AD_Org_ID IN(0,1000007) AND (wh.WH_Withholding_ID IS NULL OR wh.WH_Withholding_ID NOT IN ( SELECT Record_ID FROM AD_Private_Access WHERE AD_Table_ID = 54647 AND AD_User_ID <> 1001212 AND IsActive = 'Y' )) AND wh.IsDeclared = 'N' AND wh.IsSimulation = 'N' AND wh.DocStatus IN ('CO','CL') AND wh.Processed = 'Y' AND wh.SourceInvoice_ID IS NOT NULL AND NOT EXISTS (SELECT 1 FROM C_Invoice i WHERE wh.C_Invoice_ID = i.C_Invoice_ID AND i.DocStatus IN ('CO', 'CL', 'IP', 'IN')) AND ((wh.AD_Org_ID= ? ) AND (wh.IsManual= ? ) AND (wh.C_Currency_ID= ? ))
        at org.compiere.util.DB.getSQLValueEx(DB.java:1294)
        at org.compiere.util.DB.getSQLValueEx(DB.java:1314)
        at org.spin.service.grpc.util.db.CountUtil.countRecords(CountUtil.java:158)
        at org.spin.service.grpc.util.db.CountUtil.countRecords(CountUtil.java:64)
        at org.spin.grpc.service.ui.BrowserLogic.listBrowserItems(BrowserLogic.java:322)
        at org.spin.grpc.service.UserInterface.listBrowserItems(UserInterface.java:1264)
        at org.spin.backend.grpc.user_interface.UserInterfaceGrpc$MethodHandlers.invoke(UserInterfaceGrpc.java:1354)
        at io.grpc.stub.ServerCalls$UnaryServerCallHandler$UnaryServerCallListener.onHalfClose(ServerCalls.java:182)
        at io.grpc.PartialForwardingServerCallListener.onHalfClose(PartialForwardingServerCallListener.java:35)
        at io.grpc.ForwardingServerCallListener.onHalfClose(ForwardingServerCallListener.java:23)
        at io.grpc.ForwardingServerCallListener$SimpleForwardingServerCallListener.onHalfClose(ForwardingServerCallListener.java:40)
        at io.grpc.Contexts$ContextualizedServerCallListener.onHalfClose(Contexts.java:86)
        at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.halfClosed(ServerCallImpl.java:356)
        at io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1HalfClosed.runInContext(ServerImpl.java:861)
        at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
        at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.postgresql.util.PSQLException: ERROR: invalid reference to FROM-clause entry for table "wh"
  Hint: There is an entry for table "wh", but it cannot be referenced from this part of the query.
  Position: 217
```

--- a/grpc_utils/src/main/java/org/spin/service/grpc/util/db/CountUtil.java
+++ b/grpc_utils/src/main/java/org/spin/service/grpc/util/db/CountUtil.java
@@ -28,7 +28,7 @@ import org.compiere.util.DB;
  * @author Edwin Betancourt, EdwinBetanc0urt@outlook.com, https://github.com/EdwinBetanc0urt
  */
 public class CountUtil {
-	
+
 	/**
 	 * Count records
 	 * @param sql
@@ -73,13 +73,23 @@ public class CountUtil {
 	 * @param transactionName
 	 * @return
 	 */
-	public static int countRecords(String sql, String tableName, String tableNameAlias, List<Object> parameters, String transactionName) {
+	public static int countRecords(String originalSql, String tableName, String tableNameAlias, List<Object> parameters, String transactionName) {
 		// tableName tableName, tableName AS tableName
 		String tableWithAliases = FromUtil.getPatternTableName(tableName, tableNameAlias);
 
 		String regex = "\\s+(FROM)\\s+(" + tableWithAliases + ")"
 			+ "\\s+(\\bWHERE\\b|\\bORDER\\s+BY\\b|"
 			+ "((LEFT|INNER|RIGHT|FULL|SELF|CROSS)\\s+(OUTER\\s+){0,1}){0,1}JOIN)"
+		;
+
+		// remove comments and duplicated spaces
+		final String sql = originalSql
+			.replaceAll("/\\*.*?\\*/", "")
+			.replaceAll("--.*?\n", "")
+			.replaceAll("\\s+", " ")
+			.replaceAll("\n", "\\n")
+			.replaceAll("\r", "\\r")
+			.replace("\t", "\\t")
 		;
 
 		Pattern pattern = Pattern.compile(
@@ -122,16 +132,32 @@ public class CountUtil {
 			}
 		}
 
-		String queryCount = "SELECT COUNT(*) " + sql.substring(positionFrom, sql.length());
+		String selectColumn = "SELECT COUNT(*) ";
 
+		// verify if has DISTINCT
+		String distinctPattern = "^\\s*SELECT\\s+DISTINCT\\b";
+		boolean hasDistinct = Pattern.compile(
+			distinctPattern,
+			Pattern.CASE_INSENSITIVE | Pattern.DOTALL
+		).matcher(sql).find();
+		if (hasDistinct) {
+			// TODO: Add support with first column
+			// String firstSelectColumn = "*";
+			// selectColumn = "SELECT COUNT(DISTINCT " + firstSelectColumn + ") ";
+		}
+
+		String queryCount = selectColumn + sql.substring(positionFrom, sql.length());
 		// remove order by clause
 		queryCount = OrderByUtil.removeOrderBy(queryCount);
 
+		// Ensure parameters list is not null for `DB.getSQLValueEx`
 		if (parameters == null) {
 			parameters = new ArrayList<Object>();
 		}
 
-		return DB.getSQLValueEx(transactionName, queryCount, parameters);
+		// Execute the count query using Adempiere's DB utility.
+		int recordsLength = DB.getSQLValueEx(transactionName, queryCount, parameters);
+		return recordsLength;
 	}
 
 }

--- a/grpc_utils/src/main/java/org/spin/service/grpc/util/db/CountUtil.java
+++ b/grpc_utils/src/main/java/org/spin/service/grpc/util/db/CountUtil.java
@@ -83,14 +83,7 @@ public class CountUtil {
 		;
 
 		// remove comments and duplicated spaces
-		final String sql = originalSql
-			.replaceAll("/\\*.*?\\*/", "")
-			.replaceAll("--.*?\n", "")
-			.replaceAll("\\s+", " ")
-			.replaceAll("\n", "\\n")
-			.replaceAll("\r", "\\r")
-			.replace("\t", "\\t")
-		;
+		final String sql = SqlUtil.cleanSQL(originalSql);
 
 		Pattern pattern = Pattern.compile(
 			regex,

--- a/grpc_utils/src/main/java/org/spin/service/grpc/util/db/FromUtil.java
+++ b/grpc_utils/src/main/java/org/spin/service/grpc/util/db/FromUtil.java
@@ -23,7 +23,7 @@ import org.compiere.util.Env;
 import org.compiere.util.Util;
 
 /**
- * Class for handle SQL From
+ * Class for handle SQL FROM
  * @author Edwin Betancourt, EdwinBetanc0urt@outlook.com, https://github.com/EdwinBetanc0urt
  */
 public class FromUtil {
@@ -34,7 +34,7 @@ public class FromUtil {
 
 		MViewDefinition fromViewDefinition = viewDefinitionsList.get(0);
 		MTable table = new MTable(Env.getCtx(), fromViewDefinition.getAD_Table_ID(), null);
-		String fromClause = " FROM " + table.getTableName() + " " + fromViewDefinition.getTableAlias();
+		String fromClause = " FROM " + table.getTableName() + " AS " + fromViewDefinition.getTableAlias();
 		return fromClause;
 	}
 

--- a/grpc_utils/src/main/java/org/spin/service/grpc/util/db/JoinUtil.java
+++ b/grpc_utils/src/main/java/org/spin/service/grpc/util/db/JoinUtil.java
@@ -1,0 +1,71 @@
+/************************************************************************************
+ * Copyright (C) 2018-present E.R.P. Consultores y Asociados, C.A.                  *
+ * Contributor(s): Edwin Betancourt, EdwinBetanc0urt@outlook.com                    *
+ * This program is free software: you can redistribute it and/or modify             *
+ * it under the terms of the GNU General Public License as published by             *
+ * the Free Software Foundation, either version 2 of the License, or                *
+ * (at your option) any later version.                                              *
+ * This program is distributed in the hope that it will be useful,                  *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of                   *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                     *
+ * GNU General Public License for more details.                                     *
+ * You should have received a copy of the GNU General Public License                *
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.            *
+ ************************************************************************************/
+package org.spin.service.grpc.util.db;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Class for handle SQL JOIN
+ * @author Edwin Betancourt, EdwinBetanc0urt@outlook.com, https://github.com/EdwinBetanc0urt
+ */
+public class JoinUtil {
+
+	public static String fixSqlFromClause(String sql) {
+		String regex = "^\\s*(FROM\\s+\\S+(?:\\s*AS\\s*)?\\s+\\S+" // FROM tabla (AS) alias
+			+ "(?:\\s+(?:LEFT|INNER|RIGHT|FULL)?\\s*JOIN\\s+\\S+(?:\\s*AS\\s*)?\\s+\\S+\\s+ON\\s*\\(.*?\\))*" // zero or more JOINs
+			+ ")\\s*,\\s*(\\S+(?:\\s*AS\\s*)?\\s+\\S+)(.*)$"
+		; // Comma, table (AS) alias, and the rest
+
+		Pattern pattern = Pattern.compile(
+			regex,
+			Pattern.CASE_INSENSITIVE | Pattern.DOTALL
+		); // DOTALL to match `.` line breaks
+		Matcher matcher = pattern.matcher(sql);
+
+		if (matcher.matches()) {
+			String initialSqlPart = matcher.group(1);
+			String looseTableAlias = matcher.group(2);
+			String remainingSql = matcher.group(3);
+
+			// Separate table name and alias, optionally handling `AS`.
+			String tableName;
+			String aliasName;
+			// We use a split pattern to handle `AS` and multiple spaces
+			Pattern aliasSplitPattern = Pattern.compile("\\s*AS\\s+|\\s+", Pattern.CASE_INSENSITIVE);
+			String[] parts = aliasSplitPattern.split(looseTableAlias);
+
+			if (parts.length >= 2) {
+				tableName = parts[0].trim();
+				aliasName = parts[1].trim();
+			} else {
+				// If for some reason there is only one element (e.g. only the name of the table without alias)
+				// This is a borderline case, but it is best handled.
+				tableName = looseTableAlias.trim();
+				aliasName = tableName; // Or some default value if there is no alias
+				System.err.println("Warning: Could not parse alias for: " + looseTableAlias + ". Using the table name as alias.");
+			}
+
+			String newJoinClause = String.format("LEFT JOIN %s %s ON (1=1)", tableName, aliasName);
+
+			final String parsedSql = " " + initialSqlPart + " " + newJoinClause + remainingSql;
+			return parsedSql;
+		}
+
+		System.err.println("DEBUG: The pattern did NOT match the complete SQL. Input SQL: \n" + sql);
+		return sql; // Returns the original SQL unchanged
+	}
+
+}

--- a/grpc_utils/src/main/java/org/spin/service/grpc/util/db/SelectUtil.java
+++ b/grpc_utils/src/main/java/org/spin/service/grpc/util/db/SelectUtil.java
@@ -1,0 +1,35 @@
+/************************************************************************************
+ * Copyright (C) 2018-present E.R.P. Consultores y Asociados, C.A.                  *
+ * Contributor(s): Edwin Betancourt, EdwinBetanc0urt@outlook.com                    *
+ * This program is free software: you can redistribute it and/or modify             *
+ * it under the terms of the GNU General Public License as published by             *
+ * the Free Software Foundation, either version 2 of the License, or                *
+ * (at your option) any later version.                                              *
+ * This program is distributed in the hope that it will be useful,                  *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of                   *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                     *
+ * GNU General Public License for more details.                                     *
+ * You should have received a copy of the GNU General Public License                *
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.            *
+ ************************************************************************************/
+package org.spin.service.grpc.util.db;
+
+import java.util.regex.Pattern;
+
+/**
+ * Class for handle SQL SELECT clause and query columns
+ * @author Edwin Betancourt, EdwinBetanc0urt@outlook.com, https://github.com/EdwinBetanc0urt
+ */
+public class SelectUtil {
+
+	public static boolean isDistinct(String sql) {
+		// verify if has DISTINCT
+		String distinctPattern = "^\\s*SELECT\\s+DISTINCT\\b";
+		boolean hasDistinct = Pattern.compile(
+			distinctPattern,
+			Pattern.CASE_INSENSITIVE | Pattern.DOTALL
+		).matcher(sql).find();
+		return hasDistinct;
+	}
+
+}

--- a/grpc_utils/src/main/java/org/spin/service/grpc/util/db/SqlUtil.java
+++ b/grpc_utils/src/main/java/org/spin/service/grpc/util/db/SqlUtil.java
@@ -1,0 +1,44 @@
+/************************************************************************************
+ * Copyright (C) 2018-present E.R.P. Consultores y Asociados, C.A.                  *
+ * Contributor(s): Edwin Betancourt, EdwinBetanc0urt@outlook.com                    *
+ * This program is free software: you can redistribute it and/or modify             *
+ * it under the terms of the GNU General Public License as published by             *
+ * the Free Software Foundation, either version 2 of the License, or                *
+ * (at your option) any later version.                                              *
+ * This program is distributed in the hope that it will be useful,                  *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of                   *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                     *
+ * GNU General Public License for more details.                                     *
+ * You should have received a copy of the GNU General Public License                *
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.            *
+ ************************************************************************************/
+package org.spin.service.grpc.util.db;
+
+/**
+ * Class for handle SQL Count records
+ * @author Edwin Betancourt, EdwinBetanc0urt@outlook.com, https://github.com/EdwinBetanc0urt
+ */
+public class SqlUtil {
+	
+	/**
+     * Cleans the SQL string by:
+     * 1. Removing multi-line comments (/* ... * /).
+     * 2. Removing single-line comments (-- ...).
+     * 3. Replacing any sequence of whitespace characters (spaces, tabs, newlines, etc.) with a single space.
+     * 4. Trimming leading/trailing whitespace.
+     * This standardization simplifies subsequent regex matching and SQL parsing.
+     *
+	 * @param originalSql The original SQL string to clean.
+	 * @return A cleaned, single-line SQL string with standardized whitespace.
+	 */
+	public static String cleanSQL(String originalSql) {
+		final String sql = originalSql
+			.replaceAll("/\\*.*?\\*/", "") // Remove multi-line comments
+			.replaceAll("--.*?\n", "") // Remove single-line comments
+			.replaceAll("\\s+", " ") // Replace multiple spaces/newlines with single space
+			.trim() // Trim leading/trailing whitespace from the entire string
+		;
+		return sql;
+	}
+
+}

